### PR TITLE
expose mem-eff to autograd

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113103
* #110496
* __->__ #110495

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/functorch/test_ops.py [hunk 1]
```diff
@@ -774,6 +778,7 @@ def fn(inp, *args, **kwargs):
         skip("nn.functional.fractional_max_pool2d"),  # calls random op
         skip("nn.functional.fractional_max_pool3d"),  # calls random op
         xfail('nn.functional.scaled_dot_product_attention'),  # randomness
+        xfail('torch.ops.aten._efficient_attention_forward'),  # outputs ints
```
Comment (jbschlosser) [inline test/functorch/test_ops.py:781]: for my curiosity: why does outputting ints cause breakage

**Review by jbschlosser — APPROVED** [review]
Are CI failures irrelevant? Can we get this in? looks okay to me as long as we're okay with the BC-breaking signature changes on this private function

File: test/functorch/test_ops.py (hunk 1 above)
Comment (drisspg) [reply test/functorch/test_ops.py:781]: Ahh good question, if you grep for native functions that return integers as inputs in the code base, the only other one I can find is .item()

Alot of of our tests iterate over outputs calling methods that don't exist..

I think this gives more credence to this PR: https://github.com/pytorch/pytorch/pull/110546

That beings said I think we can still do that as followup

**Comment by drisspg** [comment]
Yup and Yup, will merge now

**Comment by drisspg** [comment]
@pytorcbot merge -f "unrelated failures"

**Comment by jeffdaily** [comment]
This has broken ROCm on trunk.  Revert or fix please.  CC @huydhn @malfet 

To verify ROCm CI prior to trunk merge you can add ciflow/rocm label.

```
2023-11-13T19:12:05.8393277Z FAILED [0.0078s] test_ops.py::TestCommonCUDA::test_out_torch_ops_aten__efficient_attention_forward_cuda_float32 - RuntimeError: USE_MEM_EFF_ATTENTION was not enabled for build.
2023-11-13T19:12:05.8394684Z 
```

**Comment by drisspg** [comment]
this requires a "skip" to be added, I will put up a forward fix

**Comment by drisspg** [comment]
@jeffdaily  fix: #110495

**Comment by malfet** [comment]
@drisspg edited (and stamped) PR number 

@jeffdaily Please note that as ROCm tests are no longer running as part of neither pull nor trunk workflows those types of regressions would be more and more frequent. There is a task to enable ROCm smoke tests on pull, but it has seen no activity in the last month or so.

**Comment by jeffdaily** [comment]
> @drisspg edited (and stamped) PR number
> 
> @jeffdaily Please note that as ROCm tests are no longer running as part of neither pull nor trunk workflows those types of regressions would be more and more frequent. There is a task to enable ROCm smoke tests on pull, but it has seen no activity in the last month or so.

@malfet rocm smoke tests are enabled now (https://github.com/pytorch/pytorch/pull/110511)

File: test/inductor/test_torchinductor_opinfo.py [hunk 2]
```diff
@@ -241,6 +241,7 @@ def format_op(op):
     "to_sparse": {f16, f32, f64},
     "pca_lowrank": {f32, f64},
     "svd_lowrank": {f32, f64},
+    "torch.ops.aten._efficient_attention_forward": {f16, bf16, f32},
```
Comment (eellison) [inline test/inductor/test_torchinductor_opinfo.py:244]: What are the failures here ? Can we fix them ?

Comment (drisspg) [reply test/inductor/test_torchinductor_opinfo.py:244]: this op returns ints and it seems like every opinfo test hates that:
https://github.com/pytorch/pytorch/pull/110546

one resolution path

**Pull request closed without merging**